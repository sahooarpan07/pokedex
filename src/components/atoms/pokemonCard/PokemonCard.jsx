import React from 'react';
import {Pressable, StyleSheet, Text, View, Image} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import {COLORS} from '../../../constants/constants';
import {getCapitalisedWord} from '../../../util/utils';
import {pokemonStyle} from './PokemonCard.styles';
const PokemonCard = ({pokemon, onPress, styles, shouldDisplayName}) => {
  const colors = pokemon?.types?.map(({type}) => COLORS[type['name']]);

  return colors.length >= 2 ? (
    <LinearGradient colors={colors} style={[pokemonStyle.grid, styles]}>
      {displayPokemonCardData(pokemon, onPress, shouldDisplayName)}
    </LinearGradient>
  ) : (
    <View style={[pokemonStyle.grid, {backgroundColor: colors[0]}, styles]}>
      {displayPokemonCardData(pokemon, onPress, shouldDisplayName)}
    </View>
  );
};

function displayPokemonCardData(pokemon, onPress, shouldDisplayName) {
  return (
    <Pressable onPress={onPress}>
      <Image
        source={{
          uri: `https://img.pokemondb.net/sprites/omega-ruby-alpha-sapphire/dex/normal/${pokemon.name}.png`,
        }}
        style={{
          height: 160,
          width: 160,
        }}
      />
      {shouldDisplayName && (
        <>
          <Text style={[pokemonStyle.name]}>
            {getCapitalisedWord(pokemon?.name)}
          </Text>
          <Text style={[pokemonStyle.name]}>
            {pokemon?.id < 10 ? '00' + pokemon?.id : '0' + pokemon.id}
          </Text>
        </>
      )}
    </Pressable>
  );
}

export default PokemonCard;
