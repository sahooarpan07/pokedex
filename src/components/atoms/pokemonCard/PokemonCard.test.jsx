/**
 * @format
 */

import 'react-native';
import React from 'react';
import PokemonCard from './PokemonCard';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders Pokemon Card correctly', () => {
  const types = [
    {
      slot: 1,
      type: {
        name: 'grass',
        url: 'https://pokeapi.co/api/v2/type/12/',
      },
    },
    {
      slot: 2,
      type: {
        name: 'poison',
        url: 'https://pokeapi.co/api/v2/type/4/',
      },
    },
  ];
  const onPress = jest.fn();
  const pokemon = {
    id: 1,
    name: 'Bulbasuar',
    types,
  };
  const styles = {flex: 1};

  const tree = renderer
    .create(<PokemonCard pokemon={pokemon} onPress={onPress} styles={styles} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
