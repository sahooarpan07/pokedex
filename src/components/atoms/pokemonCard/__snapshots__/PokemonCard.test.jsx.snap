// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pokemon Card correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4290827464,
      4291803117,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#180345",
        "borderRadius": 8,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "flex": 1,
        "justifyContent": "center",
        "margin": 16,
        "padding": 12,
        "shadowColor": "black",
        "shadowOpacity": 0.25,
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Image
      source={
        {
          "uri": "https://img.pokemondb.net/sprites/omega-ruby-alpha-sapphire/dex/normal/Bulbasuar.png",
        }
      }
      style={
        {
          "height": 160,
          "width": 160,
        }
      }
    />
  </View>
</BVLinearGradient>
`;
